pkg/config/config.go

```diff
@@ -42,103 +38,88 @@ func GetProviderFromInfrastructure(infra *configv1.Infrastructure) (configv1.Pla
 	return infra.Status.PlatformStatus.Type, nil
 }
 
-// getImagesFromImagesConfigMap collects the content of provided ConfigMap with images
-// via --images-file which is used for rendering bootstrap manifests.
-func getImagesFromImagesConfigMap(config *corev1.ConfigMap) (imagesReference, error) {
-	if config == nil || config.Data == nil {
-		return imagesReference{}, fmt.Errorf("unable to find Data field in provided ConfigMap")
+func readImagesJSONFile(filePath string) ([]byte, error) {
+	data, err := ioutil.ReadFile(filepath.Clean(filePath))
+	if err != nil {
+		return data, err
 	}
-
-	data, ok := config.Data[configMapImagesKey]
-	if !ok {
-		return imagesReference{},
-			fmt.Errorf("unable to find images key %q in ConfigMap %s", configMapImagesKey, client.ObjectKeyFromObject(config))
+	if !json.Valid(data) {
+		return nil, fmt.Errorf("images file is not a valid json")
 	}
+	return data, nil
+}
 
-	i := imagesReference{}
-	if err := json.Unmarshal([]byte(data), &i); err != nil {
-		return imagesReference{},
-			fmt.Errorf("unable to decode images content from ConfigMap %s: %v", client.ObjectKeyFromObject(config), err)
+func getCommonImageReferences(imagesFileContent []byte) (commonImagesReference, error) {
+	i := commonImagesReference{}
+	if err := json.Unmarshal(imagesFileContent, &i); err != nil {
+		return i, err
 	}
 	return i, nil
 }
 
-// getImagesFromJSONFile is used in operator to read the content of mounted ConfigMap
-// containing images for substitution in templates
-func getImagesFromJSONFile(filePath string) (imagesReference, error) {
-	data, err := ioutil.ReadFile(filepath.Clean(filePath))
-	if err != nil {
-		return imagesReference{}, err
+// ComposeConfig creates a Config for operator
+func ComposeConfig(platform configv1.PlatformType, imagesFile, managedNamespace string) (OperatorConfig, error) {
+	config := OperatorConfig{
+		Platform:         platform,
+		ManagedNamespace: managedNamespace,
 	}
 
-	i := imagesReference{}
-	if err := json.Unmarshal(data, &i); err != nil {
-		return imagesReference{}, err
+	imagesFileContent, err := readImagesJSONFile(imagesFile)
+	if err != nil {
+		klog.Errorf("Unable to decode images file from location %s", imagesFile, err)
+		return config, err
 	}
-	return i, nil
-}
 
-// getCloudControllerManagerFromImages returns a CCM binary image later used in substitution
-func getCloudControllerManagerFromImages(platform configv1.PlatformType, images imagesReference) string {
-	switch platform {
-	case configv1.AWSPlatformType:
-		return images.CloudControllerManagerAWS
-	case configv1.OpenStackPlatformType:
-		return images.CloudControllerManagerOpenStack
-	case configv1.AzurePlatformType:
-		return images.CloudControllerManagerAzure
-	default:
-		return ""
-	}
+	config.ImagesFileContent = imagesFileContent
+
+	commonImages, err := getCommonImageReferences(imagesFileContent)
+	config.CommonImages = commonImages
```

Comment:
This looks unreadable, OperatorConfig is partially initialized in the beginning, and partially here. Let's initialize in one place.